src/Bridges/bridge_optimizer.jl

```diff
@@ -1241,6 +1244,42 @@ function MOI.supports(
     )
 end
 
+function _set_substituted(
+    b::AbstractBridgeOptimizer,
+    attr::MOI.ConstraintSet,
+    ci::MOI.ConstraintIndex{F,S},
+    value,
+) where {F,S}
+    if !(value isa S)
+        throw(ArgumentError("Invalid type when setting ConstraintSet."))
+    elseif is_bridged(b, ci)
+        MOI.throw_if_not_valid(b, ci)
+        call_in_context(MOI.set, b, ci, attr, value)
+    else
+        MOI.set(b.model, attr, ci, value)
+    end
+    return
+end
+
+function _set_substituted(
+    b::AbstractBridgeOptimizer,
+    attr::MOI.ConstraintFunction,
+    ci::MOI.ConstraintIndex{F,S},
+    value,
+) where {F,S}
+    if F == MOI.VariableIndex
+        throw(MOI.SettingVariableIndexNotAllowed())
+    elseif !(value isa F)
+        throw(ArgumentError("Invalid type when setting ConstraintFunction."))
```

Comment:
Again, this should be in `throw_set_error_fallback` method so that we fix this for all model at the same time ?` In fact, I don't understand how this can be called since the signature of `MOI.set` of `AbstractBridgeOptimizer` ensures that the set has the same type.

Reply:
Because there's this fallback method for everything
https://github.com/jump-dev/MathOptInterface.jl/blob/ddbb343858245cf562da4edfc921594a734c8900/src/Bridges/bridge_optimizer.jl#L1259-L1267